spec/ruby/language/pattern_matching_spec.rb

```diff
@@ -513,6 +513,32 @@
             end
           RUBY
         }.should raise_error(SyntaxError, /illegal variable in alternative pattern/)
+
+        -> {
```

Comment:
It needs to be under a `ruby_version_is "3.2" do` guard, since it doesn't work before, and ruby/spec is run for multiple Ruby versions. A separate `it` is cleaner to highlight the behavior change in 3.2.

Reply:
Moved under the `ruby_version_is "3.2"` block.